src/py42/clients/orgs.py

```diff
@@ -1,8 +1,225 @@
 import json
 
 from py42 import settings
+from py42._internal.compat import ChainMap
 from py42.clients import BaseClient
 from py42.clients.util import get_all_pages
+from py42.exceptions import Py42Error
+from py42.settings import debug
+from py42.util import bool_required
+from py42.util import bool_to_str
+from py42.util import str_to_bool
+
+
+class OrgSettingsManager(object):
+    def __init__(self, org_client, org_id):
+        org_response = org_client.get_by_id(org_id, incSettings=True)
+        org_settings_response = org_client.get_settings_by_id(org_id)
+
+        org_dict = org_response.data
+        settings_dict = org_dict.pop("settings")
+
+        self._org = ChainMap({}, org_dict)
+        self._settings = ChainMap({}, settings_dict)
+        self._t_settings = ChainMap({}, org_settings_response.data)
+        self._org_client = org_client
+        self.errored = None
+        self.org_response = None
+        self.org_settings_response = None
+
+    @property
+    def changes(self):
+        changes_dict = {}
+        changes_dict["settings"] = self._diff_chainmap(self._org) or {}
+        changes_dict["settings"].update(self._diff_chainmap(self._settings))
```

Comment:
Is this a network call here?  Perhaps getting the changes should be a method instead of a property if this is the case. It seems to do a lot, which might be surprising if accessing as a property.

Reply:
No, this is more of a helper to be able to inspect what is actually being updated. It gets the diffs of the two dicts in each ChainMap object and only shows the keys that actually were modified. This is used both in logging the changes and in deciding whether to actually make any API calls in the `.update()` method.